sycl/include/sycl/types.hpp

```diff
@@ -910,16 +833,22 @@ template <typename Type, int NumElements> class vec {
 #error "Undefine __SYCL_BINOP macro"
 #endif
 
-#ifdef __SYCL_USE_EXT_VECTOR_TYPE__
+#ifdef __SYCL_DEVICE_ONLY__
 #define __SYCL_BINOP(BINOP, OPASSIGN, CONVERT)                                 \
   friend vec operator BINOP(const vec &Lhs, const vec &Rhs) {                  \
     vec Ret;                                                                   \
-    if constexpr (IsUsingArrayOnDevice) {                                      \
+    if constexpr (IsBfloat16 ||                                                \
+                  std::is_same_v<DataT, sycl::detail::half_impl::half>) {      \
```

Comment:
I think there is an activity to switch to native support `bfloat16`. Any chance that would help here?

Reply:
Do you mean BFloat16 support in SPIR-V? https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_bfloat16_conversion.asciidoc